doc/src/guides/solving/solve-polynomial.md

```diff
@@ -0,0 +1,515 @@
+>>> factor(p)
+(-a**2 + x)*(a**3 + a + x)
+```
+
+### Exact Numeric Solution With Root Multiplicities
+
+#### `real_roots`
+
+If the roots to your polynomial are real, using {func}`~.real_roots` ensures
+that only real (not complex or imaginary) roots will be returned.
+
+```py
+>>> real_roots(expression)
+[-2, -2, 3]
+```
+
+{func}`~.real_roots` calls {func}`~sympy.polys.rootoftools.RootOf`, so for
+equations whose roots are all real, you can get the same results by iterating
+over the number of roots of your equation:
+
+```py
+>>> [RootOf(expression, n) for n in range(0,3)]
+[-2, -2, 3]
+```
+
+### Approximate Numeric Solution With Root Multiplicities
+
+#### `nroots`
+
+{func}`~.nroots` gives an approximate numerical approximation to the roots of a
+polynomial. This example demonstrates that it can include numerical noise, for
+example a (negligible) imaginary component in what should be a real root:
+
+```py
+>>> nroots(expression)
+[3.0, -2.0 - 4.18482169793536e-14*I, -2.0 + 4.55872552179222e-14*I]
+```
+
+If you want numeric approximations of the real roots, but you want to know
+exactly which roots are real, then the best method is {func}`~.real_roots` with
+{func}`~sympy.core.evalf`:
+
+```py
+>>> [r.n(2) for r in real_roots(expression)]
+[-2.0, -2.0, 3.0]
+>>> [r.is_real for r in real_roots(expression)]
+[True, True, True]
+```
+
+{func}`~.nroots` is analogous to NumPy's {external:func}`~numpy.roots` function.
+Usually the difference between these two is that {func}`~.nroots` is more
+accurate but slower.
+
+A major advantage of {func}`~.nroots` is that it can compute numerical
+approximations of the roots of any polynomial whose coefficients can be
+numerically evaluated with {func}`~sympy.core.evalf` (that is, they do not have
+free symbols). Contrarily, symbolic solutions may not be possible for
+higher-order (fifth or greater) polynomials as explained by the [Abel-Ruffini
+theorem](https://en.wikipedia.org/wiki/Abel%E2%80%93Ruffini_theorem). Even if
+closed-form solutions are available, they may have so many terms that they are
+not useful in practice. You may therefore want to use {func}`~.nroots` to find
+approximate numeric solutions even if closed-form symbolic solutions are
+available. For example, the closed-form roots of a fourth-order (quartic)
+polynomial may be rather complicated:
+
+```py
+>>> rq0, rq1, rq2, rq3 = roots(x**4 + 3*x**2 + 2*x + 1)
+>>> rq0
+sqrt(-4 - 2*(-1/8 + sqrt(237)*I/36)**(1/3) + 4/sqrt(-2 + 7/(6*(-1/8 + sqrt(237)*I/36)**(1/3)) + 2*(-1/8 + sqrt(237)*I/36)**(1/3)) - 7/(6*(-1/8 + sqrt(237)*I/36)**(1/3)))/2 - sqrt(-2 + 7/(6*(-1/8 + sqrt(237)*I/36)**(1/3)) + 2*(-1/8 + sqrt(237)*I/36)**(1/3))/2
+```
+
+so you may prefer an approximate numerical solution:
+
+```py
+>>> rq0.n()
+-0.349745826211722 - 0.438990337475312*I
+```
+
+{func}`~.nroots` can fail sometimes for polynomials that are numerically ill
+conditioned, for example [Wilkinson's
+polynomial](https://en.wikipedia.org/wiki/Wilkinson%27s_polynomial).
```

Comment:
It is mentioned briefly below, but `RootOf.evalf` can avoid ill-conditioning, and also avoid returning spurious complex parts (see https://github.com/sympy/sympy/pull/24013#issuecomment-1237020372), because it uses a more exact, but much slower numerical algorithm based on isolating intervals.

Reply:
Is this link to the section below, what you meant?